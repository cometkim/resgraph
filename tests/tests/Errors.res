// Only records are allowed for types
@gql.type type someAbstractType

// Only valid fields can be exposed as GraphQL fields directly
@gql.type type user = {@gql.field someDict: Js.Dict.t<string>}

// Enums must be variants
@gql.enum type wrongEnum

// Enums cannot have payloads
@gql.enum type myEnum = Online | Offline | User(user)

type groupWithNoAnnotation = {@gql.field name: string}

// Unions must be variants
@gql.enum type wrongUnion

// Unions must have payloads, and they must be GraphQL object types
@gql.union type someUnion = User(user) | Group(groupWithNoAnnotation)

// Input objects must be records
@gql.inputObject type wrongInputObject

// Resolvers must be functions
@gql.field let wrongResolver = ()

// Object fields must not be promises
@gql.type type group = {@gql.field name: promise<string>}
@gql.inputObject type group2 = {name: promise<string>}

// ^ gen -- disabled for now
