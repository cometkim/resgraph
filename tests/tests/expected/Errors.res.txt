generating schema from tests/Errors.res

noticing user
Adding enum myEnum
Adding union someUnion
noticing group
Adding input object group2
Error at [1:10->1:31] in file 'Errors.res':
This type is annotated with @gql.type, but is not a record. Only records can represent GraphQL object types.

Error at [34:0->34:9] in file 'js_dict.mli':
The type `t` does not represent a valid GraphQL type.

Error at [4:34->4:42] in file 'Errors.res':
Field `someDict` is has a @gql.field annotation, but is not a valid GraphQL type. Please change the type of this field to be a valid GraphQL type, or expose this field via a custom written resolver instead if you need to transform it before exposing it in GraphQL.

Error at [7:10->7:24] in file 'Errors.res':
This type is annotated with @gql.enum, but is not a variant. Only variants can represent GraphQL enums.

Error at [10:43->10:47] in file 'Errors.res':
The variant member `User` of the GraphQL enum variant `myEnum` has a payload. Variants tagged as @gql.enum can only have members without payloads. 

Error at [15:10->15:25] in file 'Errors.res':
This type is annotated with @gql.enum, but is not a variant. Only variants can represent GraphQL enums.

Error at [12:0->12:54] in file 'Errors.res':
The type `groupWithNoAnnotation` does not represent a valid GraphQL type.

Error at [18:41->18:46] in file 'Errors.res':
The payload of the variant member `Group` of the GraphQL union variant `someUnion` is not a GraphL object. The payload needs to be a single type representing a GraphQL object, meaning it's annotated with @gql.type.

Error at [21:17->21:38] in file 'Errors.res':
This type is annotated with @gql.inputObject, but is not a record. Only records can represent GraphQL input objects.

Error at [24:15->24:28] in file 'Errors.res':
This let binding is annotated with @gql.field, but is not a function. Only functions can represent GraphQL field resolvers.

Error at [27:35->27:39] in file 'Errors.res':
This is not a valid GraphQL type: promise<string>

Error at [27:35->27:39] in file 'Errors.res':
Field `name` is has a @gql.field annotation, but is not a valid GraphQL type. Please change the type of this field to be a valid GraphQL type, or expose this field via a custom written resolver instead if you need to transform it before exposing it in GraphQL.

Error at [28:32->28:36] in file 'Errors.res':
This is not a valid GraphQL type: promise<string>

Error at [28:32->28:36] in file 'Errors.res':
Field `name` is is not a valid GraphQL type. All fields of a @gql.inputObject must be valid GraphQL types.

